email-service/src/main/java/org/java/mentorship/email/service/EmailService.java

```diff
@@ -0,0 +1,105 @@
+public class EmailService {
+
+    @Autowired
+    private JavaMailSender sender;
+
+    @Autowired
+    private Configuration config;
+
+    public EmailResponse sendTest(EmailRequest request, Map<String, Object> model) {
+
+        EmailResponse response = new EmailResponse();
+        MimeMessage message = sender.createMimeMessage();
+
+        try {
+            MimeMessageHelper helper = new MimeMessageHelper(message, MimeMessageHelper.MULTIPART_MODE_MIXED_RELATED);
+
+            Template template = config.getTemplate("email-test.ftl");
+            String html = FreeMarkerTemplateUtils.processTemplateIntoString(template, model);
+
+            helper.setTo(request.getTo());
+            helper.setFrom("budget.tracker@dfourmusic.com");
+            helper.setText(html, true);
+            helper.setSubject(request.getSubject());
+            sender.send(message);
+
+            response.setMessage("Mail sent to " + request.getTo());
+            response.setConfirmation(Boolean.TRUE);
+        } catch (MessagingException | IOException | TemplateException e) {
+            response.setMessage("Failed! " + e.getMessage());
+            response.setConfirmation(Boolean.FALSE);
+        }
+        return response;
+    }
+
+    public EmailResponse sendVerification(EmailRequest request, Map<String, Object> model) {
+
+        EmailResponse response = new EmailResponse();
+        MimeMessage message = sender.createMimeMessage();
+
+        try {
+            MimeMessageHelper helper = new MimeMessageHelper(message, MimeMessageHelper.MULTIPART_MODE_MIXED_RELATED);
+
+            Template template = config.getTemplate("email-verification.ftl");
+            String html = FreeMarkerTemplateUtils.processTemplateIntoString(template, model);
+
+            helper.setTo(request.getTo());
+            helper.setFrom("budget.tracker@dfourmusic.com");
+            helper.setText(html, true);
+            helper.setSubject(request.getSubject());
+            sender.send(message);
+
+            response.setMessage("Mail sent to " + request.getTo());
+            response.setConfirmation(Boolean.TRUE);
+        } catch (MessagingException | IOException | TemplateException e) {
+            response.setMessage("Failed! " + e.getMessage());
+            response.setConfirmation(Boolean.FALSE);
+        }
+        return response;
+    }
+
+    public EmailResponse sendOverSpend(EmailRequest request, Map<String, Object> model) {
+
+        EmailResponse response = new EmailResponse();
+        MimeMessage message = sender.createMimeMessage();
+
+        try {
+            MimeMessageHelper helper = new MimeMessageHelper(message, MimeMessageHelper.MULTIPART_MODE_MIXED_RELATED);
+
+            Template template = config.getTemplate("email-overspend.ftl");
+            String html = FreeMarkerTemplateUtils.processTemplateIntoString(template, model);
+
+            helper.setTo(request.getTo());
+            helper.setFrom("budget.tracker@dfourmusic.com");
```

Comment:
This should not be here in plaintext
Instead it should come from a property defined in the secret file